nasa/earthdata-search
# EDSC-3636: Update user order notification level preferences through CMR-ordering.

# Overview

### What is the feature?

Query to CMR-ordering to retrieve and update a user's order notification level from the Earthdata Search contact info page, replacing Legacy Services ECHO preferences and ECHO profile requests.

### What is the Solution?

Replace ECHO requests with CMR-ordering requests, get rid of references to echo_preferences and echo_profile.

### What areas of the application does this impact?

storeUserData and saveContactInfo lambdas are affected, any files that used echo_preferences, echo_profile, and echo_id  are also affected (really only files relating to contact info).

# Testing

### Reproduction steps

To test in SIT,
1. Sign in to SIT Earthdata Search and go to the [contact info page](https://search.earthdata.nasa.gov/contact_info).
2. Open your Chrome Developer Tools and go to the Network tab.
3. Change your notification level.
4. In the Network tab of your Chrome Developer Tools, click on the contact_info request and verify that under the response tab, there is an object of the form {ursId: "username", notificationLevel: "NONE"}
5. Reload the page and your updated notification level should be the visible option.
6. Also, if you open SIT CMR-ordering, updating the notification level in Earthdata Search should result in an update to the CMR-ordering database (verify with the [user query](https://git.earthdata.nasa.gov/projects/CMR/repos/cmr-ordering/browse/docs/examples/graphql/queries/user.md))

### Attachments
<img width="1720" alt="Screen Shot 2023-02-27 at 5 48 45 PM" src="https://user-images.githubusercontent.com/45649183/221704537-d7a4086c-560f-4649-bfb3-765393864f0c.png">
<img width="1728" alt="Screen Shot 2023-02-27 at 5 50 25 PM" src="https://user-images.githubusercontent.com/45649183/221704918-777c8f5f-dc2e-4680-96f0-b5cadcc84a47.png">

# Checklist
